### PR TITLE
Add examples for ORD()

### LINKED ISSUE #16941: Enrich instructions for string functions  OCTET_LENGTH() and ORD()

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #16941

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -223,7 +223,7 @@ SELECT CustomerName, CHAR_LENGTH(CustomerName) AS LenghtOfName FROM Customers;
 
 The `CHARACTER_LENGTH()` function is the same as the `CHAR_LENGTH()` function. Both functions can be used synonymously because they generate the same output.
 
-## [`CONCAT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_concat)
+### [`CONCAT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_concat)
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:226]: @Oreoxmt @qiancai this is a somewhat unrelated bugfix.

**Comment by dveeden** [comment]
Maybe we should add more info about the difference between `ORD()` and `ASCII()`?

```
mysql> SELECT ORD('e'), ORD('ë'), ASCII('e'), ASCII('ë'), HEX('e'), HEX('ë');
+----------+-----------+------------+-------------+----------+-----------+
| ORD('e') | ORD('ë')  | ASCII('e') | ASCII('ë')  | HEX('e') | HEX('ë')  |
+----------+-----------+------------+-------------+----------+-----------+
|      101 |     50091 |        101 |         195 | 65       | C3AB      |
+----------+-----------+------------+-------------+----------+-----------+
1 row in set (0.00 sec)
```

**Comment by Oreoxmt** [comment]
@dveeden Could you please invite a tech reviewer? Thanks!

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
...
+```
++----------+----------+
+| CHAR(97) | CHAR(65) |
++----------+----------+
+| a        | A        |
++----------+----------+
+1 row in set (0.01 sec)
+```
+
+In the example below you can see how `ORD()` handles multibyte characters. Here 101 and 0x65 are both the value for the "e" character, but in different formats. And 50091 and 0xC3AB are the same, but for the "ë" character.
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:1231]: Should we be more clear with the examples of `ë` is the hex representation of the UTF-8 bytes? (not the unicode code point)

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
 
 ### [`OCTET_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_octet-length)
 
-Synonym for `LENGTH()`.
+Synonym for [`LENGTH()`](#length).
 
 ### [`ORD()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ord)
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:1193]: Side note, should we still point to the MySQL documentation, when we have our own description?

**Review by mjonss — APPROVED** [review]

File: functions-and-operators/string-functions.md (hunk 3 above)
Comment (dveeden) [reply functions-and-operators/string-functions.md:1193]: As I said in https://github.com/pingcap/docs/issues/15338#issuecomment-2039767839 I think that should be done at a later stage after our own docs have been tested a bit. Then eventually we can move the links to the MySQL Reference Manual to the "See also" section.

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
...
 
 Return character code for leftmost character of the argument.
 
+This function is similar to [`CHAR()`](#char), but works the other way around.
+
+Examples:
+
+Here we take "a" and "A" and use `ORD()` to get "97" and "65".
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1201]:
```suggestion
Taking `a` and `A` as an example, `ORD()` returns `97` for `a` and `65` for `A`.
```

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
...
+```
++----------+----------+
+| ORD('a') | ORD('A') |
++----------+----------+
+|       97 |       65 |
++----------+----------+
+1 row in set (0.00 sec)
+```
+
+If we take the octal numbers we got from `ORD()` then we can get back to the orignal characters by using the `CHAR()` function. Note that the display of the output might be different if your MySQL client has the `binary-as-hex` option enabled.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1216]:
```suggestion
If you take the character code obtained from `ORD()` as input, you can get the original characters back using the `CHAR()` function. Note that the output format might vary depending on whether the `binary-as-hex` option is enabled in your MySQL client.
```

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
...
+```
++----------+----------+
+| CHAR(97) | CHAR(65) |
++----------+----------+
+| a        | A        |
++----------+----------+
+1 row in set (0.01 sec)
+```
+
+In the example below you can see how `ORD()` handles multibyte characters. Here 101 and 0x65 are both the UTF-8 encoded value for the "e" character, but in different formats. And 50091 and 0xC3AB are also the same value, but for the "ë" character.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1231]:
```suggestion
The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character. 
```

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -1188,12 +1188,61 @@ Return a string containing octal representation of a number.
...
 ### [`OCTET_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_octet-length)
 
-Synonym for `LENGTH()`.
+Synonym for [`LENGTH()`](#length).
 
 ### [`ORD()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ord)
 
 Return character code for leftmost character of the argument.
 
+This function is similar to [`CHAR()`](#char), but works the other way around.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1197]:
```suggestion
Return the character code for the leftmost character of the given argument.

This function is similar to [`CHAR()`](#char) but works the other way around.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by hfxsd — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -1188,11 +1188,60 @@ Return a string containing octal representation of a number.
...
+```
++----------+----------+
+| CHAR(97) | CHAR(65) |
++----------+----------+
+| a        | A        |
++----------+----------+
+1 row in set (0.01 sec)
+```
+
+The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1231]:
```suggestion
The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the same UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character. 
```
